### DISCUSSION TITLE
Equivalent event of FrameworkElement.Loaded, in something like Window?

### DISCUSSION THREAD

**Discussion by osamakawish** [AvaloniaUI/Avalonia#11044] [Q&A]
As stated in title, need an event equivalent to the WPF event FrameworkElement.Loaded. I'm wondering if there is any.

I've tried Initialized, but it seems the event didn't even fire. I had a debug point on VS inside the event method call, but VS never hit the debug point.

**Comment by jp2masa** [comment]
Are you using v0.10? It was added for v11 in `Control`, I think there's no simple alternative in v0.10.

**Reply by osamakawish** [reply]
Sadly yes.

**Comment by workgroupengineering** [comment]
You can try use [AttachedToVisualTree]( https://reference.avaloniaui.net/api/Avalonia/Visual/6049B47F)

**Reply by timunie** [reply]
Yeah, that's what I'd do as well. You can use `Dispatcher.UIThread.Post()` with priority `Loaded` if you need to delay the execution a bit

**Reply by maxkatz6** [reply]
For Window class it's an Opened event. Together with Dispatcher it will be a full equivalent of current implementation of Loaded event.
